pkg/proxy/access/clusteraccess.go

```diff
@@ -0,0 +1,40 @@
+package access
+
+import (
+	"net/url"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ClusterAccess holds information needed to access user namespaces in a member cluster for the specific user via impersonation
+type ClusterAccess struct { // nolint:revive
+	// APIURL is the Cluster API Endpoint for the namespace
+	apiURL url.URL
+	// impersonatorToken is a token of the impersonator's Service Account, usually the impersonator SA belongs to a member toolchaincluster
+	impersonatorToken string
+	// client is the kube client for the cluster.
+	client client.Client
```

Comment:
The client doesn't seem to be used

Reply:
Good point, removed in ce1d76fe6af282dc57a9caaed5a237f7e32b54df